packages/tools/src/types/SegmentationStateTypes.ts

```diff
@@ -66,6 +66,11 @@ export type Segmentation = {
    * inside the segmentation tools
    */
   activeSegmentIndex: number;
+  /**
+   * Preview segment index, used for drawing previews.  Will be replaced by
+   * final segment indices when completed.
+   */
+  previewSegmentIndex?: number;
```

Comment:
I'm not entirely sure about this, to be honest. I believe we should only provide a preview color, and the index should not be influenced externally. If the imported SEG already has that index, it could lead to unexpected side effects that the user may not be aware of before importing. In my opinion, we should not allow this to be controlled.

Reply:
Can do, but be aware it will be quite a lot of work, and will not preview on anything except the primary view.  Probably 3-4 days of additional effort to get this working.  Currently the code depends on being able to set the colour into a segmentation, but if that is no longer possible, then it will become significantly harder.  Probably I will implement a direct preview map to segmentation representation.  That is cleaner of course, because it means the preview can contain multiple colours and in fact only contains the new values, not the old ones, but there is a lot of code depending on the old value.

Reply:
what if the previewSegmentIndex is actually already claimed by another segment when imported? this will have side effects. Why do you think it will be 3-4 days?